numba/targets/arraymath.py

```diff
@@ -40,6 +164,67 @@ def array_sum_impl(arr):
                                     locals=dict(c=sig.return_type))
     return impl_ret_borrowed(context, builder, sig.return_type, res)
 
+@lower_builtin(np.sum, types.Array, types.int64)
+@lower_builtin("array.sum", types.Array, types.int64)
+def array_sum_axis(context, builder, sig, args):
+    """
+    The third parameter to gen_index_tuple that generates the indexing
+    tuples has to be a const so we can't just pass "axis" through since
+    that isn't const.  We can check for specific values and have 
+    different instances that do take consts.  Supporting only four
+    dimensional arrays for now.
+    """
+
+    # sum_extend defines the return type for sum with axis.  
```

Comment:
Was a typo.  Should be sum_expand.  Added a reference to where the return type is computed for sum with axis is computed.